### DISCUSSION TITLE
IQ map

### DISCUSSION THREAD

**Discussion by jaywai412** [mtex-toolbox/mtex#552] [Ask Anything] [answered]
Hello,

I have plotted IPF maps and characterized the structural properties of gb using MTEX. Thank you very much for sharing the tool.
I am trying to plot IQ maps and measure IQ values across grain boundaries; my raw data file is osc file.
Could you let me know how I should do? I could't find materials related to IQ in the archives.
Thank you.

**Comment by nyyssont** [comment, marked as answer]
Hello,

I would try

figure
plot(ebsd,ebsd.iq)
colormap gray

**Comment by kilir** [comment]
Hi,
here's a minimal example to directly get the ci (or whatever measure you have for each EBSD point) across the boundary:
```matlab
mtexdata olivine
[grains, ebsd.grainId]=ebsd.calcGrains;

% ebsd along/across boundary
ebsd_gb = ebsd('id',grains.boundary('indexed','indexed').ebsdId);

% ci along boundary
ci_gb= ebsd_gb.prop.ci

plot(ebsd_gb,ci_gb)
```
Cheers,
Rüdiger

**Comment by jaywai412** [comment]
Hi,

I have executed the following code after loading an ebsd data from osc file, which includes iq data.
I got an error saying that there is no field named as 'ci' (or 'iq').
FYI, I am using the version 5.4.0.
Thank you very much.

ebsd = EBSD.load(fname,CS,'interface','osc','convertEuler2SpatialReferenceFrame');
[grains, ebsd.grainId]=ebsd.calcGrains;
ebsd_gb = ebsd('id',grains.boundary('indexed','indexed').ebsdId);

% ci along boundary
ci_gb= ebsd_gb.prop.ci
plot(ebsd_gb,ci_gb)

**Reply by ralfHielscher** [reply]
If you type in `ebsd` into the command line it shows you which properties MTEX has found in your data. In the case of `osc` files it is probably called `ebsd.imagequality`.

Ralf.

**Comment by jaywai412** [comment]
The property name was imagequality like you said. 
Now, I can read the iq data from the file. Thank you very much.
But, there are a few undefined white data points in the plot.
I tried to fill the data using the following command as guided in the manual, but it did not work: ebsd_filled = fill(ebsd,grains);
Doesn't this function work for IQ data? 

Another thing I am trying to do is to plot IQ values along a line crossing a particular grain boundary (like a height profile).
Is there an API that can sample data along a line specified with coordinates (ebsd.x and ebsd.y)?

Thank you.

**Reply by ralfHielscher** [reply]
Hi,

it is good practice to open up new threads for follow up questions. Because it helps to keep this forum in a searchable knowledge base.

Ralf.

**Reply by ralfHielscher** [reply]
You should also mark this question as answered :)

**Reply by jaywai412** [reply]
Ralf,

I was not familiar with the github forum. 
Now, this question is marked as answered and a new thread is opened for my additional questions.

Thank you.
